---
title: "taxFun"
author:
-   name: Vinh Tran
    affiliation: Goethe University Frankfurt, Frankfurt am Main, Germany
    email: tran@bio.uni-frankfurt.de
date: "`r Sys.Date()`"
output:
    BiocStyle::html_document:
        toc_float: true
        number_sections: false
package: taxFun
abstract: |
    taxFun: A set of functions for working with NCBI taxonomy database
vignette: >
    %\VignetteIndexEntry{taxFun}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
    library(taxFun),
    collapse = TRUE,
    comment = "#>",
    dev = 'png',
    crop = NULL
)
```

# Introduction
WRITE SOMETHING HERE

# How to install taxFun

```
if (!requireNamespace("devtools"))
    install.packages("devtools")
devtools::install_github("trvinh/taxFun", INSTALL_opts = c('--no-lock'), build_vignettes = TRUE)
```

# Input and output

A text file containing list of taaxon IDs or names or a mix of them. For example:

```
Homo sapiens
100226
10090
10116
1111708
Xenopus tropicalis
122586
13616
164328
184922
```

*taxFun* will return the result in a (tab-delimited) text file. *If not specified in the command, output file will be saved as `input.*` by default.*

# Features and capabilities

| Function name | <img width=400/>Main input | <img width=400/>Output | Other options |
|---|---|---|---|
| getAllTaxonomyRanks | None | A list of all available NCBI taxonomy ranks | None |
| id2name | List of taxon IDs | A table containing input taxon IDs and their scientific names | None |
| name2id | List of taxon names | A table containing input taxon names and their taxon IDs | None |
| getRank | List of taxon IDs or/and names | A table containing input taxa and their taxonomy ranks | None |
| getRanks | List of taxon IDs or/and names | A table containing input taxon IDs, their scientific names and the names of selected taxonomy ranks | `ranks` = list of selected taxonomy ranks (Optional: if not given, all possible taxonomy ranks will be considered) |
| sortTaxonomyMatrix | List of taxon IDs or/and names | A clustered taxonomy matrix, shorted by a selected reference species | `refspec` = specified reference species |
| createTree | List of taxon IDs or/and names | Tree file in newick format | `outgroup` = outgroup species for rooting (Optional) |
| getRepresentative | List of taxon IDs or/and names | A table containing list of sub-selected taxa and their taxonomy info | `rank` = selected rank used for sub-sampling. Each supertaxon of this rank will have one representative |
| getClade | List of taxon IDs or/and names | Subset of taxa that belong to a specified taxonomy clade | `supertaxon` = clade ID or name |
| getLCA | List of taxon IDs or/and names | Last common ancestor name and ID | None |

# Examples

```{r}
# specify list of taxon IDs or taxon names
idFile <- system.file(
    "extdata", "ids.txt", package = "taxFun", mustWork = TRUE
)
nameFile <- system.file(
    "extdata", "names.txt", package = "taxFun", mustWork = TRUE
)

# convert IDs to names
taxFun("id2name", idFile)

# get names of class, family and phylum for a list of given taxon names
taxFun("getRanks", nameFile, ranks = c("class", "family", "phylum"))

# create an aligned taxonomy hierarchies for a list of taxon IDs
taxFun("sortTaxonomyMatrix", idFile, refspec = "Homo sapiens")

# create taxonomy tree rooted by an outgroup species
taxFun("createTree", nameFile, outgroup = "Homo sapiens")

# sub-sample taxa on phylum level
taxFun("getRepresentative", idFile, rank = "phylum")

# get all metazoa species
taxFun("getClade", idFile, supertaxon = "metazoa")

# get last common ancestor
taxFun("getLCA", nameFile)
```

**More examples? Please tell us what you want to see ;-)**

# How to cite
>Ngoc-Vinh Tran, Bastian Greshake Tzovaras, Ingo Ebersberger, PhyloProfile:
dynamic visualization and exploration of multi-layered phylogenetic profiles,
Bioinformatics, Volume 34, Issue 17, 01 September 2018, Pages 3041‚Äì3043,
https://doi.org/10.1093/bioinformatics/bty225

Or use the citation function in R CMD to have the citation in BibTex or
LaTeX format

```{r}
citation("taxFun")
```

# How to contribute

Thank you so much for your interest in contributing to *taxFun*! üéâüëçüçæ

Contributions to *taxFun* can take many forms. If you are

**biologist**, you can
* [report bugs](https://github.com/trvinh/taxFun/issues/new),
* [tell us about what features you would love to see](shorturl.at/dgrsO),
* improve our documentation, both
[in the Wiki](https://github.com/trvinh/taxFun/wiki) and in our
[README](https://github.com/trvinh/taxFun/blob/master/README.md),
* discuss about [non-coding issues](shorturl.at/hnDTX)

**biologist and love coding**, you can
* [fix existing bugs](https://github.com/trvinh/taxFun/issues/),
* or add new features,
* or basically whatever great idea you have!
* and all points for non-coding contributors as well :)

**not biologist but can code**, it would be great if you can
* test the tool in different environments (Windows, Linux, Mac - Firefox,
Chrome, IE, Safari,...)
* suggest a better user interface
* improve the code quality

Don't hesitate to get in touch with us if you have any questions. You can
contact us at [tran@bio.uni-frankfurt.de](mailto:tran@bio.uni-frankfurt.de)

## Contributors {.unnumbered}
- [Vinh Tran](https://github.com/trvinh)

# SessionInfo()

Here is the output of `sessionInfo()` on the system on which this document was
compiles:

```{r sessionInfo, echo = FALSE}
sessionInfo(package = "taxFun")
```

# References
